src/main/java/org/spongepowered/asm/mixin/transformer/MixinTargetContext.java

```diff
@@ -413,27 +651,28 @@ private void updateStaticBinding(MethodNode method, MethodInsnNode insn) {
      * which contains the specified method.
      * 
      * @param method Method containing the instruction
-     * @param insn INVOKE instruction node
+     * @param methodRef Unbound reference to the method
      */
-    private void updateDynamicBinding(MethodNode method, MethodInsnNode insn) {
-        this.updateBinding(method, insn, Traversal.ALL);
+    private void updateDynamicBinding(MethodNode method, MemberRef methodRef) {
+        this.updateBinding(method, methodRef, Traversal.ALL);
     }
     
-    private void updateBinding(MethodNode method, MethodInsnNode insn, Traversal traversal) {
-        if (Constants.INIT.equals(method.name) || insn.owner.equals(this.targetClass.name) || this.targetClass.name.startsWith("<")) {
+    private void updateBinding(MethodNode method, MemberRef methodRef, Traversal traversal) {
+        if (Constants.INIT.equals(method.name) || methodRef.getOwner().equals(this.targetClass.name) || this.targetClass.name.startsWith("<")) {
             return;
         }
         
-        Method superMethod = this.targetClassInfo.findMethodInHierarchy(insn.name, insn.desc, traversal == Traversal.ALL, traversal);
+        Method superMethod = this.targetClassInfo.findMethodInHierarchy(methodRef.getName(), methodRef.getDesc(), traversal == Traversal.ALL,
+                traversal);
         if (superMethod != null) {
             if (superMethod.getOwner().isMixin()) {
-                throw new InvalidMixinException(this, "Invalid " + ASMHelper.getOpcodeName(insn) + " in " + this + " resolved " + insn.owner
-                        + " -> " + superMethod.getOwner() + " for " + insn.name + insn.desc);
+                throw new InvalidMixinException(this, "Invalid " + ASMHelper.getOpcodeName(methodRef.getOpcode()) + " in " + this + " resolved "
```

Comment:
Maybe define `toString()` in `MemberRef` to return something prettier and then just use that.